.github/workflows/build-test-distribute.yaml

```diff
@@ -30,6 +28,42 @@ jobs:
           make clean
       - run: |
           make check
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: "Check if run on releasable ref"
+        id: "releasable_ref_checker"
+        if: |
+          (github.ref == 'refs/branch/master') ||
+          startsWith(github.ref, 'refs/branch/release-')  ||
+          startsWith(github.ref, 'refs/tags/') ||
+          false
```

Comment:
would `github.event_name == "push"` be easier?  It also solves any problem with changing releasable branches in the future.

I think you can just have the 2 steps merged together:

```
if: github.event_name == "push" || contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix')
```

WDYT?

Reply:
Yes, it is better. In this way, we narrow down the maintainance to one place